Modules/_ssl.c

```diff
@@ -645,11 +645,11 @@ PySSL_SetError(PySSLSocket *sslsock, int ret, const char *filename, int lineno)
         {
             if (e == 0) {
                 PySocketSockObject *s = GET_SOCKET(sslsock);
-                if (ret == 0 || (((PyObject *)s) == Py_None)) {
+                if (((PyObject *)s) == Py_None) {
                     p = PY_SSL_ERROR_EOF;
                     type = state->PySSLEOFErrorObject;
                     errstr = "EOF occurred in violation of protocol";
-                } else if (s && ret == -1) {
+                } else {
```

Comment:
Should not it be?
```suggestion
                } else if (s) {
```

Reply:
@serhiy-storchaka 
Yes, you are right. I was mistaken that Py_None would be the same as NULL. I will revise as your advice.